src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.js

```diff
@@ -0,0 +1,65 @@
+import React, { useRef, useState, useEffect } from "react";
+import ReactDOM from "react-dom";
+import { connect } from "react-redux";
+import { actions } from "ui/actions";
+import { selectors } from "ui/reducers";
+import "./LineNumberTooltip.css";
+
+export function StaticTooltip({ targetNode, children }) {
+  const { top, left } = targetNode.getBoundingClientRect();
+  const docWidth = document.querySelector("html").getBoundingClientRect().width;
+
+  return ReactDOM.createPortal(
+    <div className="static-tooltip" style={{ top: `${top}px`, right: `${docWidth - left}px` }}>
+      {children}
+    </div>,
+    targetNode
+  );
+}
+
+export function LineNumberTooltip({ editor, cx, runAnalysisOnLine, analysisPoints }) {
+  const [lineNumberNode, setLineNumberNode] = useState(null);
+
+  const setHoveredLineNumber = event => {
+    let target = event.target;
+    const isBreakpointMarkerNode = target.closest(".new-breakpoint");
+
+    // If hovered on a breakpoint marker, get the corresponding linenumber element.
+    if (isBreakpointMarkerNode) {
+      const gutterNode = target.closest(".Codemirror-gutter-elt");
+      target = gutterNode?.previousElementSibling;
+    }
```

Comment:
could we move this logic to gutter-events too so we just emit the line number? This is cute, but I don't think we want to think about other gutter markers in this api, just the gutter el and line


```
const setHoveredLineNumber = (line, target) => {
    runAnalysisOnLine(cx, line);
    setLineNumberNode(target);
}
```